### PR TITLE
Project-labyrinth: Erika A Porath och Marielle Ohlsson

### PR DESCRIPTION
**Pull request by Marielle8**
Link to my deployed project: https://book-labyrinth.netlify.app/

My reflections on how this week's project turned out: turned out good but we didn't have time to style it as much as we wanted to. We aimed for the blue requirments of this project.

Things I'd like to have clarified or explained in more detail:

### CODE REVIEW

File: code/src/components/GameScreen.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+  const dispatch = useDispatch()
+
+  return (
+    <div className='gamescreen-container'>
+      {loading &&
+        <span className='image-container'>
+          <img src={assets} alt='compass' />
+        </span>
+      }
```
Comment (Polkapetunian) [inline code/src/components/GameScreen.js:21]: Cute loading spinner! =) Nice idea and very fitting to the game.

File: code/src/components/GameScreen.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      {!loading && (
+        <div className='description-container'>
+          <p className='fade-in'>{gameStart.description}</p>
```
Comment (Polkapetunian) [inline code/src/components/GameScreen.js:24]: The fade in effect of the text is a nice touch!

File: code/src/components/GameScreen.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+              <p className='fade-in'>{action.description}</p>
+              <button
+                className='fade-in'
+                onClick={() => dispatch(nextAction(action.direction))}
+              >
+                Go {action.direction}
+              </button>
+            </div>
+          ))}
+          {gameStart.coordinates === '1,3' && <Summary />}
```
Comment (Polkapetunian) [inline code/src/components/GameScreen.js:36]: Smart idea to render the summary based on the coordinates.

File: code/src/index.css [hunk 4]
```diff
@@ -1,13 +1,223 @@
+@import url(//db.onlinewebfonts.com/c/e3619f7413527cff15063e007b666239?family=WrittenLies);
```
Comment (Polkapetunian) [inline code/src/index.css:1]: This font is tricky to read and should have more weight and bigger size. Especially the footer is impossible to read in current size. This error shows in the inspect tool: Mixed Content: The page at 'https://book-labyrinth.netlify.app/' was loaded over HTTPS, but requested an insecure font 'http://db.onlinewebfonts.com/t/e3619f7413527cff15063e007b666239.woff2'. This request has been blocked; the content must be served over HTTPS.

File: code/src/index.css [hunk 5]
```diff
@@ -1,13 +1,223 @@
...
+
 body {
   margin: 0;
   font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
     "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
     sans-serif;
   -webkit-font-smoothing: antialiased;
   -moz-osx-font-smoothing: grayscale;
+  background-color:  #E5D5C0;
+  background-image: url('./assets/book-background.jpg');
```
Comment (Polkapetunian) [inline code/src/index.css:23]: The background looks good in both mobile and desktop. Nice theme!

File: code/src/index.css [hunk 6]
```diff
@@ -1,13 +1,223 @@
...
+
+/* Animation */
+.fade-in {
+  animation: fadeIn ease 3s;
+  -webkit-animation: fadeIn ease 3s;
+  -moz-animation: fadeIn ease 3s;
+  -o-animation: fadeIn ease 3s;
+  -ms-animation: fadeIn ease 3s;
+}
+
```
Comment (Polkapetunian) [inline code/src/index.css:169]: This adds a lot to the experience!

File: code/src/utils/urls.js [hunk 7]
```diff
@@ -0,0 +1,2 @@
+export const API_START = 'https://wk16-backend.herokuapp.com/start'
+export const API_ACTION = 'https://wk16-backend.herokuapp.com/action'
```
Comment (Polkapetunian) [inline code/src/utils/urls.js:2]: Neat!

**Review by Polkapetunian — APPROVED** [review]
Solid work! You have made it fully functional and the code is clean and easy to read. The idea with the book is really nice, especially with the fade-in-effect. The font needs some more attention. It would have been nice to see more styling since it seems that you have interesting ideas.

File: code/src/components/GameScreen.js (hunk 3 above)
Comment (KarolCislowski) [reply code/src/components/GameScreen.js:36]: It is a nice solution, but it is not a future-proof one. What if the map structure will change on the backend (the game will grow for example)? However, in this case, we can safely assume that we are in the final room if there are **no remaining actions**

So our logic could be something like:

```jsx
if (actions.length === 0) {
   return <GameOver />;
}else{
...
```
or 
```jsx
...
actions.length === 0 && <EndScreen/>
...
```

File: code/src/components/StartScreen.js [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+        value={inputValue}
+        onChange={event => setInputValue(event.target.value)}
+      />
+      <button
+        onClick={onStartSet}
+        disabled={!inputValue}
+      >
+        Start the story
+      </button>
+    </div>
```
Comment (KarolCislowski) [inline code/src/components/StartScreen.js:31]: It is not nice that you forgot about the form tag. It striped your input from submission on enter, what is the most used sending the form method 😭

**Review by KarolCislowski — APPROVED** [review]
Great app 🥳

**Pull request closed without merging**